reactive/docker.py

```diff
@@ -207,13 +207,42 @@ def container_sdn_setup(sdn):
     if data_changed('bip', bind_ip) or data_changed('mtu', mtu):
         status_set('maintenance', 'Configuring container runtime with SDN.')
         opts = DockerOpts()
-        # TODO pop the value if it exists.
+        # This is a great way to misconfigure a docker daemon. Remove the
+        # existing bind ip and mtu values of the SDN
+        if opts.exists('bip'):
+            opts.pop('bip')
+        if opts.exists('mtu'):
+            opts.pop('mtu')
         opts.add('bip', bind_ip)
         opts.add('mtu', mtu)
-        _reconfigure_docker_for_sdn()
+        _remove_docker_network_bridge()
         set_state('docker.sdn.configured')
 
 
+@when_not('sdn-plugin.available')
+@when('docker.sdn.configured')
+def scrub_sdn_config():
+    ''' If this scenario of states is true, we have likely broken a
+    relationship to our once configured SDN provider. This necessitates a
+    cleanup of the Docker Options for BIP and MTU of the presumed dead SDN
+    interface. '''
+
+    opts = DockerOpts()
+    try:
+        opts.pop('bip')
+        opts.pop('mtu')
+    except KeyError:
+        hookenv.log('Unable to locate bip or mtu in Docker config.')
+        hookenv.log('Assuming no action required, and restarting the daemon.')
+
```

Comment:
hmmm, seems odd that you would have an interface MTU configuration without BIP.  But i concede the point, that its possible a prior execution could have unintended side effects with checking both inline like this instead of independent checks.